### PR TITLE
Fix `AlignModelTest` tests

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

- Fix `AlignModelTest` torchscript tests.  This model has non-persistent buffer, and need a bit more care (same as in common test)
```python
        self.register_buffer(
            "token_type_ids", torch.zeros(self.position_ids.size(), dtype=torch.long), persistent=False
        )
```

- Fix `AlignModelTest.test_multi_gpu_data_parallel_forward`: same as CLIP, we need a even number for `batch_size`, as this model have `logits_per_image` and `logits_per_text` which is of shape `(batch_size, batch_size)`, and in order to gather across devices, we need the second dim being the same.

### CODE REVIEW

File: tests/models/align/test_modeling_align.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ class AlignVisionModelTester:
     def __init__(
         self,
         parent,
-        batch_size=13,
+        batch_size=12,
```
Comment (ydshieh) [inline tests/models/align/test_modeling_align.py:68]: Same as CLIP, due to the fact that `logits_per_text` and `logits_per_image` being of shape (batch_size, batch_size).

**Review by sgugger — APPROVED** [review]
Thanks for fixing!
